From 282a65bb157b524484bca44b5237b0024ee183c0 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 10 Jan 2025 15:09:17 +0100
Subject: [PATCH 1546/1695] dts: stm32: fix support for I2C3 on stm32mp215f-dk

Configure I2C3 pins during sleep mode in board device-tree file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 04ca37dc891d6ad005c7c805985e49239bb10260 dts: stm32: fix support for I2C3 on stm32mp215f-dk]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id5cd0724326c467b0f8b2aa936d93c649f190e79
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/428151
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp21-pinctrl.dtsi | 7 +++++++
 core/arch/arm/dts/stm32mp215f-dk.dts     | 3 ++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
index 1186bdcf2..c05654ac1 100644
--- a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
@@ -61,4 +61,11 @@
 			slew-rate = <0>;
 		};
 	};
+
+	i2c3_sleep_pins_a: i2c3-sleep-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('Z', 1, ANALOG)>, /* I2C3_SCL */
+				 <STM32_PINMUX('Z', 0, ANALOG)>; /* I2C3_SDA */
+		};
+	};
 };
diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index 7749db6ad..6e1fa9114 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -56,8 +56,9 @@
 };
 
 &i2c3 {
-	pinctrl-names = "default";
+	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c3_pins_a>;
+	pinctrl-1 = <&i2c3_sleep_pins_a>;
 	i2c-scl-rising-time-ns = <185>;
 	i2c-scl-falling-time-ns = <20>;
 	clock-frequency = <400000>;
-- 
2.47.2

